src/awkward/operations/ak_where.py

```diff
@@ -87,57 +87,57 @@ def _impl1(condition, mergebool, highlevel, behavior):
 
 
 def _impl3(condition, x, y, mergebool, highlevel, behavior):
-    akcondition = ak.operations.to_layout(
+    behavior = behavior_of(condition, x, y, behavior=behavior)
+    backend = backend_of(condition, x, y, default=cpu, coerce_to_common=True)
+
+    condition_layout = ak.operations.to_layout(
         condition, allow_record=False, allow_other=False
-    )
+    ).to_backend(backend)
 
-    left = ak.operations.to_layout(x, allow_record=False, allow_other=True)
-    right = ak.operations.to_layout(y, allow_record=False, allow_other=True)
+    x_layout = ak.operations.to_layout(x, allow_record=False, allow_other=True)
+    if isinstance(x_layout, ak.contents.Content):
+        x_layout = x_layout.to_backend(backend)
 
-    good_arrays = [akcondition]
-    if isinstance(left, ak.contents.Content):
-        good_arrays.append(left)
-    if isinstance(right, ak.contents.Content):
-        good_arrays.append(right)
-    backend = backend_of(*good_arrays, default=cpu)
```

Comment:
I think this was here to avoid passing non backend-aware objects to `backend_of`. This is not necessary any more (and hasn't been for a while), so it's OK to remove this.

Reply:
I'm 90% sure you're right about that being the intention of this code.